src/Hng.Domain/Entities/Role.cs

```diff
@@ -6,6 +6,10 @@ public class Role : EntityBase
         public string Description { get; set; }
         public bool IsActive { get; set; }
         public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
-        public DateTime UpdatedAt { get; set; }
+        public DateTime? UpdatedAt { get; set; } = null;
+        public  Guid OrganizationId { get; set; }
+        public Organization Organisation { get; set; }
+        public ICollection<RolePermission> Permissions { get; set; } = [];
+        public ICollection<UserRole> UsersRoles { get; set; } = [];
```

Comment:
There is no point having this here. Once you add a collection of Roles in User and collection of Users in Role, EntityFramework creates the link table.

Reply:
organisations need to know their users roles